From: Takashi Iwai <tiwai@suse.de>
Subject: Workaround for missing HD-audio on AMD platforms
Patch-mainline: Never, ever; temporary workaround
References: bsc#1202492

The recent change to add support for IOMMU group in PCI core
(commit 512881eacfa7 "bus: platform,amba,fsl-mc,PCI: Add device DMA
ownership management") caused a regression at probing HD-audio devices
on some devices like AMD platforms; it results in the lack of complete
HD-audio devices.

Until the issue gets fixed properly in IOMMU and/or PCI core side,
apply a temporay workaround in HD-audio driver.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/pci/hda/hda_intel.c |    1 +
 1 file changed, 1 insertion(+)

--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -2793,6 +2793,7 @@ static struct pci_driver azx_driver = {
 	.driver = {
 		.pm = AZX_PM_OPS,
 	},
+	.driver_managed_dma = true, // XXX: FIXME - just a workaround
 };
 
 module_pci_driver(azx_driver);
